ui/components/src/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo } from 'react';
+import { UnitOptions, formatValue } from '@perses-dev/core';
+import { use, EChartsCoreOption } from 'echarts/core';
+import { BarChart as EChartsBarChart } from 'echarts/charts';
+import { GridComponent, DatasetComponent, TitleComponent, TooltipComponent } from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useTheme } from '@mui/material';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import { EChart } from '../EChart';
+
+use([EChartsBarChart, GridComponent, DatasetComponent, TitleComponent, TooltipComponent, CanvasRenderer]);
+
+export interface BarChartData {
+  label: string;
+  value: number | null | undefined;
+}
+
+export interface BarChartProps {
+  width: number;
+  height: number;
+  data: BarChartData[] | null | undefined;
+  unit: UnitOptions;
+}
+
+export function BarChart(props: BarChartProps) {
+  const { width, height, data, unit } = props;
+  const chartsTheme = useChartsTheme();
+  const theme = useTheme();
+
+  const option: EChartsCoreOption = useMemo(() => {
+    if (data == null) return chartsTheme.noDataOption;
+
+    const source: Array<Array<BarChartData['label'] | BarChartData['value']>> = [];
+    data.map((d) => {
+      source.push([d.label, d.value]);
+    });
+
+    return {
+      title: {
+        show: false,
+      },
+      dataset: [
+        {
+          dimensions: ['label', 'value'],
+          source: source,
+        },
+      ],
+      xAxis: {},
+      yAxis: {
+        type: 'category',
+        splitLine: {
+          show: false,
+        },
+      },
+      series: {
+        type: 'bar',
+        label: {
+          position: 'right',
+          show: true,
+          formatter: (params: { data: number[] }) => params.data[1] && formatValue(params.data[1], unit),
+          color: theme.palette.text.primary,
+        },
+        itemStyle: {
+          borderRadius: 4,
+          color: theme.palette.designSystem.blue[400],
+        },
+      },
+      tooltip: {
```

Comment:
i think we might have to implement a custom tooltip here similar to what we have for time series chart so we can pin the tooltip 

@sjcobb what do you think?

Reply:
I’m not sure if tooltip pinning is really as useful on BarChart, it’s more relevant on TimeSeriesChart since there can be so many series (pin to scroll, click ‘Show All’, potentially filter in future, etc.).

I was hoping we could use ECharts tooltip and style it to look as close to the custom React TimeSeriesTooltip as possible (and make those styles inherited in chartsTheme so they’re applied by default anytime ECharts tooltip is used)

Really the only reason I built the custom React tooltip in the first place was for performance when there are hundreds or thousands of series, the ECharts tooltip is pretty great in all other cases and gives embed users a lot more freedom to customize options, adjust the formatter, decide if they want trigger axis or trigger item, etc.

Reply:
(Or change the TimeSeriesTooltip styles to match if we end up liking how the ECharts tooltip looks better)

But yeah, we should confirm with @foxbat07, how we integrate with tracing in the future (do we need a way to display a context menu?) will also play a factor

Reply:
+1 I don't think we need a custom tooltip since it's not focused on timeseries data. However, I would be in favor of getting styling close as possible. I would say do a fast follow it we can't do it quickly

Reply:
I'll look into moving the tooltip styling into the echarts theme and work with Mohit to make sure all tooltip styling is the same!

Reply:
I updated the tooltip styling to match the `TimeSeriesTooltip` and moved it into the echarts theme 6d10815c5ef38cc9db53a5a2ad7724792faa85e2
<img width="184" alt="Screen Shot 2023-07-10 at 1 38 40 PM" src="https://github.com/perses/perses/assets/22085207/43b3f9f2-e83d-4829-ba1f-c18bf6275420">